remove ps_product_country_tax from db_structure.sql

MajinBakar opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project!

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/#pull-requests

For type and category see:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#type--category
------------------------------------------------------------------------------->

| Questions         | Answers |
|-------------------|---------|
| **Branch?**           | develop |
| **Description?**      | Removes the creation of the `ps_product_country_tax` table from `install-dev/data/db_structure.sql`.This table has been present in the database structure for many versions of PrestaShop (at least since 1.6), but it is never used in the core, nor in native modules. It does not appear in any queries, logic, or schema relationships across the codebase.Its removal helps clean up the database schema and reduces confusion for developers and database administrators. S/o ChDUP. Also, remove the reference to this table on several files (S/o PrestaEdit)
| **Type?**             | refacto |
| **Category?**         | CO |
| **BC breaks?**        | no |
| **Deprecations?**     | no |
| **How to test?**      | 1. Run a fresh install using `install-dev` directory.<br>2. After installation, connect to the database and confirm that the table `ps_product_country_tax` is no longer present.<br>3. Navigate through the back office and front office to confirm everything works normally. |
| **UI Tests**          | N/A (no user interface impact) |
| **Fixed issue or discussion?** | N/A (can open discussion if needed) |
| **Related PRs**       | N/A |
| **Sponsor company**   | Friends of Presta  |

MajinBakar committed:
remove ps_product_country_tax from db_structure.sql

Removes the creation of the `ps_product_country_tax` table from `install-dev/data/db_structure.sql`.

This table has been present in the database structure for many versions of PrestaShop.

PrestaEdit reviewed (changes requested):
There is another part of the source code where it's used ; more to restore/truncate the table but it's need to be removed too.

Hlavtox commented:
@PrestaEdit Good catch

@MajinBakar 
![Snímek obrazovky 2025-04-24 104006](https://github.com/user-attachments/assets/04910b2b-c77a-46cf-b659-02a38dd2f4c1)

MajinBakar committed:
$_product_country_tax removed from Tax.php

 $_product_country_tax removed from Tax.php. The reference table is unused.

MajinBakar committed:
Truncate table product_country_tax  removed from AdminImportController.php

Truncate table product_country_tax  removed from AdminImportController.php. The table is unused.

MajinBakar committed:
product_country_tax removed from ImportEntityDeleter.php

product_country_tax removed from ImportEntityDeleter.php. The reference table is unused.

ps-jarvis commented:
<!-- PR_TABLE_DESCRIPTION_ERROR -->
Hi, thanks for this contribution!

I found some issues with the Pull Request description:

* The `branch` should be `develop`, `9.0.x` or `8.2.x`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#branch))
* The `description` shouldn&#039;t be empty. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#description))
* The `type` should be one of these: `new feature`, `improvement`, `bug fix` or `refacto`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#branch))
* The `category` should be one of these: `FO`, `BO`, `CO`, `IN`, `WS`, `TE`, `LO`, `ME` or `PM`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#branch))
* The `BC breaks` should be `yes` or `no`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#bc-breaks))
* The `deprecations` should be `yes` or `no`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#deprecations))
* The `How to test` shouldn&#039;t be empty. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#how-to-test))

Would you mind having a look at it? This will help us understand how interesting your contribution is, thank you very much!

(Note: this is an automated message, but answering it will reach a real human)

MajinBakar committed:
product_country_tax removed from ProductResetter.php

product_country_tax removed from ProductResetter.php. The reference table is unused.

MajinBakar committed:
product_country_tax removed from changes.php

product_country_tax removed from changes.php. The reference table is unused.

MajinBakar committed:
product_country_tax removed from relations.php

product_country_tax removed from relations.php. The reference table is unused.

kpodemski commented:
Thank you @MajinBakar ! 

Even though it prevents the creation of a database table,  I think having it in PrestaShop 9.1 is safe if the table hasn't been used for a long time.

@MajinBakar, if you want to include this contribution in PrestaShop 9.0, you will have to use the `9.0.x` branch. What do you think? Is 9.1 okay, or do you want to rebase/re-create the PR against the `9.0.x` branch?
